app/controllers/api/base_controller/authentication.rb

```diff
@@ -30,6 +30,10 @@ def require_api_user_or_token
         log_api_auth
       end
 
+      def api_config
+        @api_config ||= VMDB::Config.new("vmdb").config[Settings.base.module.to_sym] || {}
```

Comment:
Actually now that I think about, why have the confusing abstractions with this method as well as the `Settings.base.module` bit?  Settings for the api are under `::Settings.api`, so anywhere you have api_config currently you can just replace that directly with `::Settings.api`, and I think it would be MUCH simpler to read.


For example...

```ruby
authenticate_options = {
  :timeout      => api_config[:authentication_timeout].to_i_with_method
}

# vs

authenticate_options = {
  :timeout      => ::Settings.api.authentication_timeout.to_i_with_method
}
```

To me the latter is much more clearly expressing that this value comes directly from the user configuration.


Incidentally, the whole `Settings.base.module` feels so strange to even have.  I mean, I can understand if there was a thought it might change, but like many things, YAGNI.  (I can't imagine it ever changing anyway...all clients would have to change, which would be painful)